### PR TITLE
provider/openstack: Refactored upsert load balancer to use lbaasv2

### PR DESCRIPTION
**Pull request by derekolk**

### CODE REVIEW

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/BlockingClientAdapter.groovy [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    this.statusChecker = statusChecker
+    this.timeout = timeout
+    this.pollInterval = pollInterval
+  }
+
+  static BlockingClientAdapter from (StatusChecker s) {
+    new BlockingClientAdapter(s, TimeUnit.SECONDS.toMillis(60), TimeUnit.SECONDS.toMillis(5))
+  }
+
+  static BlockingClientAdapter from (long timeout, long pollInterval, StatusChecker s) {
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/BlockingClientAdapter.groovy]: Any reason to have static factory methods instead of just the two constructors?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        .protocol(ListenerProtocol.forValue(externalProtocol))
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  ListenerV2 getListener(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().listener().get(id)
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Do you want to ensure the listener is found, like `getLoadBalancer`?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        .protocol(Protocol.forValue(internalProtocol))
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  LbPoolV2 getPool(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().lbPool().get(id)
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Also ensure the pool is found

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+      getRegionClient(region).networking().port().get(portId)
+    }
+  }
+
+  @Override
+  Port updatePort(final String region, final String portId, final List<String> securityGroups) {
+    handleRequest {
+      // Builder doesn't take in list of security groups and doesn't allow you to set the ID so, adding some ugly code :)
+      PortBuilder portBuilder = Builders.port()
+      securityGroups.each { portBuilder = portBuilder.securityGroup(it) }
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Are you replacing `portBuilder` each time or is it just adding all the security groups to the port builder?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/loadbalancer/OpenstackLoadBalancerDescription.groovy [hunk 5]
```diff
@@ -16,23 +16,34 @@
...
-import com.netflix.spinnaker.clouddriver.openstack.domain.LoadBalancerMethod
-import com.netflix.spinnaker.clouddriver.openstack.domain.LoadBalancerProtocol
+import com.netflix.spinnaker.clouddriver.openstack.domain.HealthMonitor
 import groovy.transform.AutoClone
 import groovy.transform.Canonical
 
 @AutoClone
 @Canonical
 class OpenstackLoadBalancerDescription extends OpenstackAtomicOperationDescription {
-  String region
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/loadbalancer/OpenstackLoadBalancerDescription.groovy:28]: Do you not need region anymore?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 6]
```diff
@@ -16,203 +16,391 @@
...
-    newLoadBalancerPool.setCreatedTime(System.currentTimeMillis())
+      if (listenersToDelete) {
+        List<ListenerV2> deleteValues = existingListenerMap.findAll { listenersToDelete.contains(it.key) }.collect {
+          it.value
+        }
+        removeListenersAndPools(region, resultLoadBalancer.id, deleteValues)
+      }
 
-    if (!openstackClientProvider.getSubnet(region, subnetId)) {
-      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Unable to retrieve referenced subnet ${subnetId} in ${region}."
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy:93]: A lot of `task.updateStatus`s have been removed. Do you want to add a couple back where it makes sense?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+   */
+  ActionResponse deleteMonitor(final String region, final String id)
+
+  /**
+   * Retreives port by id.
+   * @param region
+   * @param portId
+   * @return
+   */
+  Port getPort(final String region, final String portId)
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: This belongs in the networking provider.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+  /**
+   * Updates port by id.
+   * @param region
+   * @param portId
+   * @param securityGroups
+   * @return
+   */
+  Port updatePort(final String region, final String portId, final List<String> securityGroups)
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Same

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+  /**
+   * Associates floating ip address to port.
+   * @param region
+   * @param floatingIpId
+   * @param portId
+   * @return
+   */
+  NetFloatingIP associateFloatingIpToPort(final String region, final String floatingIpId, final String portId)
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Same

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+
+  /**
+   * Disassociates floating ip address from port.
+   * @param region
+   * @param floatingIpId
+   * @return
+   */
+  NetFloatingIP disassociateFloatingIpFromPort(final String region, final String floatingIpId)
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Same

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+        .maxRetries(healthMonitor.maxRetries)
+        .timeout(healthMonitor.timeout)
+        .urlPath(healthMonitor.url)
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  Port getPort(final String region, final String portId) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: This belongs in the networking provider.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+    handleRequest {
+      getRegionClient(region).networking().port().get(portId)
+    }
+  }
+
+  @Override
+  Port updatePort(final String region, final String portId, final List<String> securityGroups) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+      PortBuilder portBuilder = Builders.port()
+      securityGroups.each { portBuilder = portBuilder.securityGroup(it) }
+      Port changedPort = portBuilder.build()
+      changedPort.setId(portId)
+      getRegionClient(region).networking().port().update(changedPort)
+    }
+  }
+
+  @Override
+  NetFloatingIP associateFloatingIpToPort(final String region, final String floatingIpId, final String portId) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+    handleRequest {
+      getRegionClient(region).networking().floatingip().associateToPort(floatingIpId, portId)
+    }
+  }
+
+  @Override
+  NetFloatingIP disassociateFloatingIpFromPort(final String region, final String floatingIpId) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 15]
```diff
@@ -16,203 +16,391 @@
...
-    }
 
-    task.updateStatus UPSERT_LOADBALANCER_PHASE, "Done upserting load balancer ${resultPool?.name} in ${region}"
-    return [(region): [id: resultPool?.id]]
-  }
+      Map<String, ListenerV2> existingListenerMap = buildListenerMap(region, resultLoadBalancer)
+      Map<String, Listener> listenerMap = description.listeners.collectEntries([:]) {
+        [(getListenerKey(it.externalPort, it.externalProtocol.name(), it.internalPort, it.internalProtocol.name())): it]
+      }
+      Map<String, ListenerV2> listenersToUpdate = Maps.newHashMap()
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: use `[:]` for consistency

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 16]
```diff
@@ -16,203 +16,391 @@
...
+      Map<String, Listener> listenersToAdd = Maps.newHashMap()
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: use `[:]` for consistency

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 17]
```diff
@@ -16,203 +16,391 @@
...
-    } catch (OpenstackProviderException ope) {
-      throw new OpenstackOperationException(AtomicOperations.UPSERT_LOAD_BALANCER, ope)
-    }
 
-    task.updateStatus UPSERT_LOADBALANCER_PHASE, "Done upserting load balancer ${resultPool?.name} in ${region}"
-    return [(region): [id: resultPool?.id]]
-  }
+      Map<String, ListenerV2> existingListenerMap = buildListenerMap(region, resultLoadBalancer)
+      Map<String, Listener> listenerMap = description.listeners.collectEntries([:]) {
+        [(getListenerKey(it.externalPort, it.externalProtocol.name(), it.internalPort, it.internalProtocol.name())): it]
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: name closure param

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 18]
```diff
@@ -16,203 +16,391 @@
...
+        }
+      }
 
-  protected LbPool createLoadBalancer(String region, String subnetId, LoadBalancerPool newLoadBalancerPool, VirtualIP virtualIP, PoolHealthMonitor poolHealthMonitor) {
-    LbPool resultPool
-    OpenstackClientProvider openstackClientProvider = getClientProvider()
+      Set<String> listenersToDelete = Sets.difference(existingListenerMap.keySet(), Sets.union(listenersToAdd.keySet(), listenersToUpdate.keySet()))
 
-    newLoadBalancerPool.setCreatedTime(System.currentTimeMillis())
+      if (listenersToDelete) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: unnecessary null check - change to check if size > 0

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 19]
```diff
@@ -16,203 +16,391 @@
...
+          it.value
+        }
+        removeListenersAndPools(region, resultLoadBalancer.id, deleteValues)
+      }
 
-    if (!openstackClientProvider.getSubnet(region, subnetId)) {
-      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Unable to retrieve referenced subnet ${subnetId} in ${region}."
-      throw new OpenstackOperationException(AtomicOperations.UPSERT_LOAD_BALANCER, "Subnet ${subnetId} not found in ${region}")
-    }
+      if (listenersToAdd) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: unnecessary null check - change to check if size > 0

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 20]
```diff
@@ -16,203 +16,391 @@
...
+        addListenersAndPools(region, resultLoadBalancer.id, description.name, description.algorithm, listenersToAdd, description.healthMonitor)
+      }
 
-    task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating lbPool ${newLoadBalancerPool.name} in ${region}..."
-    resultPool = openstackClientProvider.createLoadBalancerPool(region, newLoadBalancerPool)
-    virtualIP.poolId = resultPool.id
-    task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created lbPool ${newLoadBalancerPool.name} in ${region}"
+      if (listenersToUpdate) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: unnecessary null check - change to check if size > 0

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 21]
```diff
@@ -16,203 +16,391 @@
...
+   * Creates a load balancer in given subnet.
+   * @param region
+   * @param name
+   * @param subnetId
+     * @return
+     */
+  protected LoadBalancerV2 createLoadBalancer(String region, String name, String subnetId) {
+    task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating load balancer $name in ${region} ..."
+    String description = generateCreatedTime(System.currentTimeMillis())
+    LoadBalancerV2 result = BlockingClientAdapter.from { LoadBalancerV2 loadBalancer ->
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: This is a bit confusing - I was expecting the from to be the operation to execute, and the execute (check) to be the check to run on. Flipping them might make it more clear.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 22]
```diff
@@ -16,203 +16,391 @@
...
-    OpenstackClientProvider openstackClientProvider = getClientProvider()
+  /**
+   * Finds all of the stacks/server groups that the given load balancer is associated with and updates the stack defintion
+   * so pool members will be added/removed with create/upsert changes.
+   * @param region
+   * @param loadBalancerId
+   * @param memberDataList
+   */
+  protected void updateStacks(String region, String loadBalancerId) {
+    clientProvider.listStacks(region).each {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: Update this to list stacks based on filtering params - `["tags": loadBalancerId]`

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 23]
```diff
@@ -16,203 +16,391 @@
...
+   * Adds/Removes floating ip address associated to load balancer.
+   * @param region
+   * @param networkId
+   * @param portId
+   */
+  protected void updateFloatingIp(String region, String networkId, String portId) {
+    NetFloatingIP existingFloatingIp = clientProvider.getFloatingIpForPort(region, portId)
+    if (networkId) {
+      Network network = clientProvider.getNetwork(region, networkId)
+      FloatingIP ip = clientProvider.getOrCreateFloatingIp(region, network.name)
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy:220]: Shouldn't we be using NetFloatingIP? That is how floating ips are cached.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 24]
```diff
@@ -16,203 +16,391 @@
...
+   * Adds listeners and pools to an existing load balancer.
+   * @param region
+   * @param loadBalancerId
+   * @param name
+   * @param listeners
+   * @return
+   */
+  protected void addListenersAndPools(String region, String loadBalancerId, String name, Algorithm algorithm, Map<String, Listener> listeners, HealthMonitor healthMonitor) {
+    BlockingClientAdapter pollingClientAdaptor = getPollingClientAdaptor(region, loadBalancerId)
+    listeners?.entrySet()?.each { Map.Entry<String, Listener> entry ->
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: just do

```
listeners?.each { k,v ->
...
}
```

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 25]
```diff
@@ -16,203 +16,391 @@
...
+  /**
+   * Adds/Removes/Updates a health monitor to a given load balancer and pool.
+   * @param region
+   * @param loadBalancerId
+   * @param lbPool
+     */
+  protected void updateHealthMonitor(String region, String loadBalancerId, LbPoolV2 lbPool, HealthMonitor healthMonitor) {
+    BlockingClientAdapter pollingClientAdaptor = getPollingClientAdaptor(region, loadBalancerId)
+    if (lbPool.healthMonitorId) {
+      if (healthMonitor) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: Is this check needed?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 4 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: This is adding all security groups to the port builder.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/loadbalancer/OpenstackLoadBalancerDescription.groovy (hunk 5 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/loadbalancer/OpenstackLoadBalancerDescription.groovy:28]: region and account are refactored into base description object.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 6 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy:93]: I think I added enough back in ... There are 34 within the operation.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 2 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Yep ... I am really starting to despise this design choice by having to wrap all calls in try/catch blocks.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 3 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Yep

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy (hunk 7 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy (hunk 8 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy (hunk 9 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy (hunk 10 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 11 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 12 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 13 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy (hunk 14 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Same. Agree will move.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 15 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 17 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 19 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 20 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 21 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: As per conversation, will rename to BlockingStatusChecker

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 24 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 22 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: I am going to wait to do this in a separate PR because I need to change the server group create as well.  Added TODO.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 18 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 16 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: done.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 23 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy:220]: Looks identical to the current solution -> https://github.com/spinnaker/clouddriver/blob/master/clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy#L181-L210
